invenio_communities/models.py

```diff
@@ -1,151 +1,99 @@
 # General Public License for more details.
 #
 # You should have received a copy of the GNU General Public License
-# along with Invenio; if not, write to the Free Software Foundation, Inc.,
-# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
-
-
-"""Community models.
-
-A layer around Invenio's collections and portalboxes for easier allowing
-end-users to create their own collections.
-
-    from invenio_communities.models import Community
-    from invenio_ext.sqlalchemy import db
-
-    u = Community.query.get('test5')
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
 
-    u = Community(
-        id='test5',
-        id_user=176,
-        title='European Middleware Initiative',
-        description='Bla bla',
-        curation_policy='Bla'
-    )
-    db.session.add(u)
-    db.session.commit()
-    u.save_collection(provisional=False)
+"""Invenio-Communities database models."""
 
-After call to save_collection() you must do the following:
-- Clear redis cache key for collection
-- Run webcoll immediately for only this collection
-"""
+from __future__ import absolute_import, print_function
 
 from datetime import datetime
 
-from flask import url_for
-
-from invenio_base.globals import cfg
-from invenio.config import CFG_SITE_LANG
-from invenio_ext.sqlalchemy import db
-from invenio_ext.template import render_template_to_string
-from invenio.legacy.bibrecord import record_add_field
-from invenio_access.models import \
-    AccACTION, AccARGUMENT, \
-    AccAuthorization, AccROLE, UserAccROLE
 from invenio_accounts.models import User
-from invenio_communities.signals import before_save_collection, \
-    after_save_collection, before_save_collections, after_save_collections, \
-    before_delete_collection, after_delete_collection, \
-    before_delete_collections, after_delete_collections, \
-    pre_curation, post_curation
-from invenio_collections.models import \
-    Collection, \
-    CollectionCollection, \
-    CollectionFormat, \
-    CollectionPortalbox, \
-    Collectiondetailedrecordpagetabs, \
-    Collectionname, \
-    Portalbox
-from invenio_records.api import get_record
-from invenio_oaiharvester.models import OaiREPOSITORY
-
-
-class Community(db.Model):
-
-    """Represent a Community.
-
-    A layer around Invenio's collections and portalboxes.
-    """
+from invenio_db import db
+from invenio_records.models import RecordMetadata
+from sqlalchemy_utils.models import Timestamp
+from sqlalchemy_utils.types import UUIDType
 
-    __tablename__ = 'community'
 
-    #
-    # Fields
-    #
-    id = db.Column(db.String(100), primary_key=True)
+class InclusionRequest(db.Model, Timestamp):
```

Comment:
Hm, I think its better to include the inclusion request inside the record, since then it can be easily searched using all the existing functionality we have. Alternatively we have to amend the indexed record in elasticsearch. We can talk more when I'm back on monday.